docs/17-common-problems-and-their-solutions.md

```diff
@@ -4,97 +4,22 @@ Common problems and their solutions
 
-## EOFException when using client classes
-
-An exception of the following form might be due to a change in schema:
-
-```bash
-Caused by: java.io.EOFException
-     at java.io.DataInputStream.readFully(DataInputStream.java:197)
-     at java.io.DataInputStream.readUTF(DataInputStream.java:609)
-     at java.io.DataInputStream.readUTF(DataInputStream.java:564)
-Exception in thread "main" sleeper.core.statestore.StateStoreException: Exception querying DynamoDB
-```
-
-When cdk deploy is used to deploy a table, the state store is initialised. This initialisation writes the initial
-partitions to the partitions part of the state store (if using the default DynamoDB state store then this will be a
-table named `sleeper-<instance-id>-table-<table-name>-partitions`). The partitions are specific to the schema of the
-table (specifically to the row keys specified). Therefore if you deploy a table with a certain schema, and then attempt
-to change the schema by updating the table properties file, any interactions with the partitions table will not be able
-to read the partitions.
-
-If you want to reinitialise a table with a new schema, see the section below on reinitialising a table.
-
 ## Out of memory error from standard ingest tasks
 
 If standard ingest tasks fail with an out of memory error ("Exception in thread main java.lang.OutOfMemoryError: Java
 heap space") then this is likely due to the tasks not being able to store the specified number of records in memory.
-Standard ingest works by reading a certain number of records (given by `sleeper.ingest.memory.max.batch.size`) into
+Standard ingest works by reading a certain number of records (given by `sleeper.ingest.arrow.working.buffer.bytes`) into
 memory. These are sorted and then written to a local file. This process is repeated a certain number of times until a
-certain number of records in total (given by `sleeper.ingest.max.local.records`) have been written to local disk.
-Sensible values for the parameters `sleeper.ingest.memory.max.batch.size` and `sleeper.ingest.max.local.records`
+certain number of records in total (given by `sleeper.ingest.arrow.max.single.write.to.file.records`) have been written to local disk.
+Sensible values for the parameters `sleeper.ingest.arrow.max.local.store.bytes` and `sleeper.ingest.arrow.batch.buffer.bytes`
 obviously depend on the data - the more fields the schema has and the bigger those fields are, the more space will be
 used and the fewer records will fit into memory / on disk.
 
-If you see an out of memory error, then try reducing `sleeper.ingest.memory.max.batch.size`. When reducing this
-parameter it is a good idea to also reduce `sleeper.ingest.max.local.records`. To change these parameters, use the
+If you see an out of memory error, then try reducing `sleeper.ingest.arrow.max.local.store.bytes`. When reducing this
+parameter it is a good idea to also reduce `sleeper.ingest.arrow.batch.buffer.bytes`. To change these parameters, use the
```

Comment:
I don't think this makes sense any more? When configuring the maximum number of records, you don't know exactly how much memory this will take, so it makes sense that you might need to reduce it. When you configure the amount of space directly it will either fit in memory or not.

It's also not true that the local store size would have an effect on out of memory errors. The working buffer an batch buffer are in memory, while the local store is on disk.